core/src/main/java/io/grpc/internal/DelayedStream.java

```diff
@@ -298,10 +325,11 @@ public void run() {
         }
       });
     } else {
+      drainPendingCalls();
       if (listenerToClose != null) {
```

Comment:
Seems `listenerToClose` must be nonnull here (after adding assertion `checkState(listener != null)`). `delegateToRealStream  == true` iff `listenerToClose == null` as they change value together.

Reply:
Yes, `listener` is non-null and it is copied to `listenerToClose` only when `delegateToRealStream = false`. Removed `listenerToClose`, as there's no longer any synchronization necessary when accessing listener.